_posts/2015-03-23-design-patterns-part-1.markdown

```diff
@@ -0,0 +1,142 @@
+---
+layout: post
+title: "Design Patterns, Fast Tests, and Robust Software, Part I"
+date: 2015-03-23 12:01:00
+categories: design patterns, testing, software architecture
+---
+
+# Design Patterns, Fast Tests, and Robust Software, Part I
+
+Writing high quality software means expecting change: changes to business requirements, changes to the web framework you’ve chosen, changes to the
+libraries you depend on and changes to the coding language.  At Sittercity, we typically rely on Rails for our web framework, but our Rails projects don't looks like most Rails projects.  When we're building a Rails app, we try to leverage the good parts of Rails and avoid the bad.  We typically construct views and controllers in a conventional way, but handle persistent storage much differently than an idiomatic rails app.
```

Comment:
Consider removing typically from "We typically construct views and..." too many typicallys :D